Content.Server/Medical/DefibrillatorSystem.cs

```diff
@@ -112,6 +113,16 @@ public bool CanZap(EntityUid uid, EntityUid target, EntityUid? user = null, Defi
         if (!component.CanDefibCrit && _mobState.IsCritical(target, mobState))
             return false;
 
+        if (HasComp<CMDefibrillatorBlockedComponent>(target))
+        {
+            if (TryComp(target, out CMDefibrillatorBlockedComponent? block))
+            {
+                if (user != null)
+                    _popup.PopupEntity(Loc.GetString(block.Popup), uid, user.Value);
+                return false;
+            }
+        }
```

Comment:
You're checking the same thing twice here, TryComp also does a HasComp implicitly.
```suggestion
        if (TryComp(target, out CMDefibrillatorBlockedComponent? block))
        {
            if (user != null)
                _popup.PopupEntity(Loc.GetString(block.Popup), uid, user.Value);
            return false;
        }
```